staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2125,6 +2125,10 @@ type ScreenshotOptions struct {
 	MoveCursor bool `json:"moveCursor"`
 }
 
+type VSOCKOptions struct {
```

Comment:
ym, yes vsock could go intoo its own group. Would allow graduation ...

Reply:
Do you mean move the `VSOCKOptions` to `core/v1alpha`? Shall I also change the REST URL to `v1alpha`?

Reply:
Yes, I think both should be in v1alpha

Reply:
So all of `core` is on `v1`. If we want to give it a `v1alpha1` we have to put it into its own api group. Not sure if we want to do that here, but that is for instance how it works for snapshots, ...

Reply:
I only see downside in saying this is v1 when we have a clear path to graduate an API.  Alpha buys us flexibility and some time to run this in production before we make up our minds.

Reply:
We did not have it so far for subresources. Having an extra api group for every new subresource call we add may be a little bit too much. It is behind a feature gate. That is right now the typical way, similar to adding api fields to the v1 api ...

Reply:
I agree. The current graduation process is for resources not subresources.

Reply:
Subresource or not, I still consider this a user facing API because the type can be vendored by clients.  The graduation process for subresources will be easier because it's only clients that are affected.  So we may not currently have a process for graduating subresources but I think we should :).

Reply:
Subresources are attached to their objects. We have this `subresources.kubevirt.io` group due to technical reasions (I would have preferred to not have it), but typically new subresources end up in k8s exactly on the object they belong to. If that object has already reached `v1` then it appears there. 

Fine-tuning graduation processes in general when we talk about feature-gated code definitely makes sense. But not sure if we should start that within that PR :)

Reply:
Here's what I can live with.  While we can consider changing the policy away from subresources always being v1, this vsock API must have a release note if there are breaking changes so it's clear that clients need to upgrade.  Something like `[update kubevirt clients]` in the message will do.

Reply:
> Here's what I can live with. While we can consider changing the policy away from subresources always being v1

I would prefer to stick here with k8s, which does the same. Wen the version of an api group gets bumped, the old version may stick around but is typically not extended anymore. :/